aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,324 @@
-.. include:: /../common/stub-topic.txt
+Filters
+=======
 
-|stub-icon| Filters
-===================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+*Filters* in ASP.NET MVC allow you to run code before or after a particular stage in the execution pipeline. Filters can be configured globally, per-controller, or per-action.
 
-.. _issue: https://github.com/aspnet/Docs/issues/119
+.. contents:: Sections
+    :local:
+    :depth: 1
+
+`View or download sample from GitHub <https://github.com/aspnet/Docs/tree/master/mvc/controllers/filters/sample>`_.
+
+How do filters work?
+--------------------
+
+Each filter type is executed at a different stage in the pipeline, and thus has its own set of intended scenarios. Choose what type of filter to create based on the task you need it to perform, and where in the request pipeline it executes. Filters run within the MVC Action Invocation Pipeline, sometimes referred to as the *Filter Pipeline*, which runs after MVC selects the action to execute.
+
+.. image:: filters/_static/filter-pipeline-1.png
+
+Different filter types run at different points within the pipeline. Some filters, like authorization filters, only run before the next stage in the pipeline, and take no action afterward. Other filters, like action filters, can execute both before and after other parts of the pipeline execute, as shown below.
+
+.. image:: filters/_static/filter-pipeline-2.png
+
+Selecting a Filter
+^^^^^^^^^^^^^^^^^^
+
+:ref:`Authorization filters <authorization-filters>` are used to determine whether the current user is authorized for the request being made.
+
+:ref:`Resource filters <resource-filters>` are the first filter to handle a request after authorization, and the last one to touch the request as it is leaving the filter pipeline. They're especially useful to implement caching or otherwise short-circuit the filter pipeline for performance reasons.
+
+:ref:`Action filters <action-filters>` wrap calls to individual action method calls, and can manipulate the arguments passed into an action as well as the response returned from it.
```

Comment:
`as well as the response returned from it` -> I'd mention that this is about the action result returned from the action.

Generally dealing with the 'response' is something the action result will do